### ISSUE TITLE
closeTab should present a confirmation dialog

### ISSUE THREAD

**Issue by carlos-zamora** [microsoft/terminal#5301]
# Description of the new feature/enhancement
`closeTab` should present a confirmation dialog if multiple panes are open. This would prevent users from accidentally closing all panes in a tab by accident.

**Comment by eugenesvk** [comment]
I'd suggest to have this user-configurable so that you could have both a close-without-confirmation command and a close-with-confirmation one and bind them to separate keybinds if you like (e.g. with the latter command bound to combo less fat-finger prone)

**Comment by htcfreek** [comment]
I had the same request in the linked issue. One suggestion I'd like to mention here is that you should ask the user what to do instead of simply warning the user.

### Example
**Tab with split-pane:** powershell | cmd
**Question:** 
```
What should be closed?
Closing the last tab closes the whole window.

[*this pane*] [tab] [cancel]
```

**Comment by smuuf** [comment]
This would be so great to have.

**Comment by Vampire** [comment]
Please also make it possible to show a warning on simple tab closing, not only with multiple panes.
It happened multiple times to me that I accidentally closed a tab and sometimes there were even processes running inside.
I'd prefer to always be warned before closing a tab.

**Comment by zadjii-msft** [comment]
Yep, I'm pretty sure that's tracked as a part of https://github.com/microsoft/terminal/issues/6549#issuecomment-675494526.

**Comment by tovine** [comment]
Same here, I just misclicked and lost one of my tabs with an active SSH connection and ongoing work.
At least ask for confirmation if a process (like SSH) is running, closing an "idle" terminal is OK but one that is in use is sad :cry:

**Comment by nihsal** [comment]
Is this still being implemented? Its been 5 years and it's still a major issue especially on lower resolution displays with multiple tabs open. The close button takes 50% of the clickable space.

**Comment by psychob** [comment]
I learned - the hard way - that tab title can resize itself, and then magically i won't click on the visible edge. I click on the close the tab button. Yay, progress lost.

**Comment by fourpastmidnight** [comment]
I can't tell you _HOW MANY TIMES_ recently I've inadvertently clicked that damn little X when "switching tabs", closing my console with lots of valuable history or terminating my SSH session to a machine I'm working on! Why isn't this yet in the product?

@DHowett-MSFT I saw the closed PR for this, and I understand that there are some decisions that need to be made. But this is clearly a _HUGE_ pain point for many users. Is this something the team can prioritize soon+ As others have said, WT is 5 years old and the code in the PR to make this happen is quite simple--so if all that's blocking this is "governance" of settings, can't we get that prioritized and close this pain point in the near future (like, in weeks, not _years_)?

Thanks for your consideration.

**Comment by fourpastmidnight** [comment]
> Same here, I just misclicked and lost one of my tabs with an active SSH connection and ongoing work. At least ask for confirmation if a process (like SSH) is running, closing an "idle" terminal is OK but one that is in use is sad 😢

Even closing an "idle" terminal may not be OK. I have long running PowerShell sessions with lots of valuable history in them as I'm working through things--and I've accidentally clicked that damn tiny littel "X" when I intended to switch tabs.....

**Comment by Vampire** [comment]
Just switch off the X and you don't accidentally click it ;-)

**Comment by fourpastmidnight** [comment]
> Just switch off the X and you don't accidentally click it ;-)

I see no such option in the Settings pages to disable the 'x' on individual console tabs.

**Comment by Vampire** [comment]
It's described in one of the related issues at least.
Can be, that you have to do it in the settings json file.
It is an option of the theme where you can conigure `showCloseButton` to `never` for `tab`, then you don't have that pesky button.

**Comment by WolffM** [comment]
hey, @zadjii-msft I saw that in #19944 you're going for a more comprehensive confirmCloseOn enum. But given that the design is still being discussed, would it make sense to land a simple fix now and iterate from there?

I went for a simple fix to resolve this issue. It adds a simple confirmCloseAllPanes boolean setting and a confirmation dialog when closing a tab with multiple panes.

curious to hear what you think.
